e2e/specs/edit-form-with-interactive-builder.spec.ts

```diff
@@ -0,0 +1,257 @@
+                questionOptions: {
+                  rendering: 'text',
+                  concept: 'a-system-defined-concept-uuid',
+                },
+                id: 'sampleQuestion',
+              },
+            ],
+          },
+        ],
+      },
+    ],
+    processor: 'EncounterFormProcessor',
+    referencedForms: [],
+    uuid: 'xxx',
+    version: '1',
+    description: 'This is test description',
+  };
+
+  await test.step('When I visit the form builder', async () => {
+    await formBuilderPage.gotoFormBuilder();
+  });
+
+  await test.step('And I search for the form I need to edit', async () => {
+    await formBuilderPage.searchForForm(form.name);
+  });
+
+  await test.step('And I click the `Edit` button on the form I need to edit', async () => {
+    await formBuilderPage.page.getByRole('row', { name: form.name }).getByLabel('Edit Schema').first().click();
+  });
+
+  await test.step('And then I click the `Interactive Builder` tab', async () => {
+    await formBuilderPage.interactiveBuilderTab().click();
+  });
+
+  await test.step('Then I click the `Edit` button on the form name I need to edit', async () => {
+    await formBuilderPage.editFormButton().click();
+  });
+
+  await test.step('And then I fill in the updated form name', async () => {
+    await formBuilderPage.editFormNameInput().click();
+    await formBuilderPage.editFormNameInput().fill('An edited form');
+  });
+
+  await test.step('And then I click the `save` button', async () => {
+    await formBuilderPage.saveButton().click();
+  });
+
+  await test.step('Then I should get a success message and the form name should be renamed', async () => {
+    await expect(formBuilderPage.page.getByText(/form renamed/i)).toBeVisible();
+    await expect(formBuilderPage.page.getByRole('heading', { level: 1, name: /an edited form/i })).toBeVisible();
+    const formTextContent = await formBuilderPage.schemaEditorContent().textContent();
+    expect(JSON.parse(formTextContent)).toEqual(updatedForm);
+  });
+
+  await test.step('Then I click the `Edit` button on the page I need to edit', async () => {
+    await formBuilderPage.editPageButton().click();
+  });
+
+  await test.step('And then I fill in the updated page name', async () => {
+    await formBuilderPage.editPageNameInput().fill('An edited page');
+    updatedForm.pages[0].label = 'An edited page';
+  });
+
+  await test.step('Then I click the `Save` button', async () => {
+    await formBuilderPage.saveButton().click();
+  });
+
+  await test.step('Then I should get a success message and the page name should be renamed', async () => {
+    await expect(formBuilderPage.page.getByText(/page renamed/i)).toBeVisible();
+    await expect(formBuilderPage.page.getByRole('heading', { level: 1, name: /an edited page/i })).toBeVisible();
+    const formTextContent = await formBuilderPage.schemaEditorContent().textContent();
+    expect(JSON.parse(formTextContent)).toEqual(updatedForm);
```

Comment:
Opted to use `toEqual` since it should check the entire object to make sure unintentional changes to other questions, etc wasn't made. The funtion `toMatch` that you previously used didn't achieve that.